### DISCUSSION TITLE
How to handle received Contact from request Contact ReplyButton ?

### DISCUSSION THREAD

**Discussion by evrard-c** [def-studio/telegraph#181] [Q&A] [answered]
This is a **Button** on my **Keyboard**.
`Button::make('English')->action('selectLocale')->param('locale', 'en')`
If the user press it, the _**selectLocale**_ method on my _**CustomWebhookHandler**_ will take the action to select English as the user language.

Then, this is a **ReplyButton** for a Contact request on my **ReplyKeyboard**.  
`ReplyButton::make('Send Contact') ->requestContact()`
How could my _**CustomWebhookHandler**_ handle the received data on a specific method like _**contactSave**_ ?
Generally, how could i handle data from ReplyButtons on a specific method?

**Comment by fabio-ivona** [comment, marked as answer]
Hi! 

you can handle inside your CustomWebhookHandler by using the handleChatMessage method:

```php
class CustomWebhookHandler extends WebhookHandler
{
    //....

    protected function handleChatMessage(Stringable $text): void
    {
        $phone = $this->message->contact()->phoneNumber();
        //...
    }
}
```

**Reply by evrard-c** [reply]
Thanks for the answer @fabio-ivona. 
That's what I was doing, I needed to be sure that it was the right way to do it. 😉
